### PR TITLE
RavenDB-5648 proper implementation for index creation

### PR DESCRIPTION
**Pull request by ppekrol**

### CODE REVIEW

File: src/Raven.Client/Connection/Operation.cs [hunk 1]
```diff
@@ -25,15 +29,24 @@ public Operation(AsyncServerClient asyncServerClient, long id)
...
+            {
+                await _asyncServerClient.changes.Value.ConnectionTask.ConfigureAwait(false);
+                var observableWithTask = _asyncServerClient.changes.Value.ForOperationId(_id);
+                _subscription = observableWithTask.Subscribe(this);
+                await FetchOperationStatus().ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                _result.TrySetException(e);
+                throw;
```
Comment (ayende) [inline src/Raven.Client/Connection/Operation.cs]: Should we throw here? No one is going to ever catch that error.

File: src/Raven.NewClient/Connection/Operation.cs [hunk 2]
```diff
@@ -26,15 +29,24 @@ public Operation(AsyncServerClient asyncServerClient, long id)
...
+            {
+                await _asyncServerClient.changes.Value.ConnectionTask.ConfigureAwait(false);
+                var observableWithTask = _asyncServerClient.changes.Value.ForOperationId(_id);
+                _subscription = observableWithTask.Subscribe(this);
+                await FetchOperationStatus().ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                _result.TrySetException(e);
+                throw;
```
Comment (ayende) [inline src/Raven.NewClient/Connection/Operation.cs]: Same

**Review by ayende — APPROVED** [review]

File: src/Raven.Client/Connection/Operation.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Client/Connection/Operation.cs]: I know, but thought that it would be better to set task state to faulty?

Comment (ayende) [reply src/Raven.Client/Connection/Operation.cs]: The problem is that then this raised the UnhandledTaskException, and we get bug reports on that.

Comment (ppekrol) [reply src/Raven.Client/Connection/Operation.cs]: removed 'throw'

**Pull request merged**